api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```
@@ -0,0 +1,251 @@
+        design.setReportDefinition(reportDefinition);
+        design.setRendererType(CsvReportRenderer.class);
+
+        // TODO: should we parameterize these? this was stuff I just copied from Mirebaials... are we okay with the default being ISO-8859-1; not sure what the blacklist does
+        design.addPropertyValue("blacklistRegex", "[^\\p{InBasicLatin}\\p{L}]");
+        design.addPropertyValue("characterEncoding", "ISO-8859-1");
+        design.addPropertyValue("dateFormat", "dd-MMM-yyyy HH:mm:ss");
+
+        return design;
+    }
+
+    public static List<ReportDescriptor> loadReportDescriptors() {
+        List<ReportDescriptor> reportDescriptors = new ArrayList<ReportDescriptor>();
+        Collection<File> files = null;
+
+        try {
+            File reportDir = new File(getReportingDescriptorsConfigurationDir());
+            if (reportDir.exists()) {
+                // search all directories and subdirectories for YAML files
+                files = FileUtils.listFiles(reportDir, FileFilterUtils.suffixFileFilter("yml"), TrueFileFilter.INSTANCE);
+            }
+        }
+        catch (Exception e) {
+            log.error("Unable to open reporting configuration directory", e);
+        }
+
+        if (files != null) {
+            for (File file : files) {
+                reportDescriptors.add(ReportLoader.load(file));
+            }
+        }
+
+        return reportDescriptors;
+    }
+
+    public static ReportDescriptor load(File file) {
+
+        ReportDescriptor reportDescriptor = null;
+
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+
+        if (file.exists()) {
+
+            InputStream is = null;
+
+            try {
+                is = new FileInputStream(file);
+                reportDescriptor = objectMapper.readValue(is, ReportDescriptor.class);
+            }
+            catch (Exception e) {
+                log.error("Unable to load report descriptor " + file.getAbsolutePath(), e);
```

Another place where I think we should fail hard.